crates/router/src/connector/utils.rs

```diff
@@ -130,6 +135,31 @@ where
             _ => self.status,
         }
     }
+    fn get_amount_capturable<F>(&self, payment_data: &PaymentData<F>) -> Option<i64>
+    where
+        F: Clone,
+    {
+        let intent_status =
+            IntentStatus::foreign_from(self.get_attempt_status_for_db_update(payment_data));
+        match intent_status {
+            IntentStatus::RequiresCapture => Some(payment_data.payment_attempt.get_total_amount()),
+
+            IntentStatus::Processing
+            | IntentStatus::RequiresCustomerAction
+            | IntentStatus::RequiresMerchantAction
+            | IntentStatus::RequiresPaymentMethod
+            | IntentStatus::RequiresConfirmation
+            // terminal statuses
+            | IntentStatus::PartiallyCaptured
+            | IntentStatus::Succeeded
+            | IntentStatus::Failed
+            | IntentStatus::Cancelled => Some(0),
```

Comment:
Why are we returning 0 incase of Processing, RequiresCustomerAction, RequiresPaymentMethod, RequiresConfirmation?

RequiresCustomerAction - Incase of 3ds manual capture, while going for capture amount capturable will become 0 and capture will fail.

RequiresPaymentMethod, RequiresConfirmation - These are initial status for the payment intent incase of manual capture again the above scenario will happen.

Processing - If Payment attempt has AuthenticationPending, Authorizing, VoidInitiated, CaptureInitiated, Pending, Started, AuthenticationSuccessful, CodInitiated all are mapped to Processing status which again fail capture call for manual capture payment

Reply:
A payment can be captured only when the intent status is `requires_capture`. Payment can have `RequiresCustomerAction, RequiresPaymentMethod, RequiresConfirmation, Processing` before the status changes to `requires_capture`. In All other cases, `amount_capturable` should be 0 since the payment cannot be captured. Once the status changes to `requires_capture`, we will set `amount_capturable` as `total_authorized_amount()`.

Reply:
> A payment can be captured only when the intent status is `requires_capture`. Payment can have `RequiresCustomerAction, RequiresPaymentMethod, RequiresConfirmation, Processing` before the status changes to `requires_capture`. In All other cases, `amount_capturable` should be 0 since the payment cannot be captured. Once the status changes to `requires_capture`, we will set `amount_capturable` as `total_authorized_amount()`.

This solution was found to be not feasible. So Instead of deciding amount_capturable based only on intent_status. We should calculate amount_capturable based on intent status as well as the payment flow. Have made this change.